### DISCUSSION TITLE
Error while generating metric plots: Train_loss metric storing 70 values instead of 35"

### DISCUSSION THREAD

**Discussion by koch-tobias** [manujosephv/pytorch_tabular#269] [Q&A]
I am currently facing a bit of an issue where my trained model's results are not being stored correctly for one specific metric.
My code is generating metric plots using the function 'get_model_results' which uses 'results_from_experiment' to gather results from the trained model for each epoch.

However, for the metric 'train_loss', 70 values are being stored instead of the expected 35 values. All other metrics are being stored correctly with 35 values. I am wondering if anyone has faced a similar issue before and if so, how did you solve it?

I would really appreciate any insights or suggestions that could help me resolve this issue.

Thank you!

```
def results_from_experiment(event_path, metric):
    results_list = []
    for e in tf.compat.v1.train.summary_iterator(event_path):
        for v in e.summary.value:
            if v.tag == metric:
                results_list.append(v.simple_value)
    
    return results_list
 ```
```
def get_model_results(event_path):

    result_dict = { 
                    "train_loss": [],
                    "train_accuracy": [], 
                    "train_f1_score": [],
                    "valid_loss": [],
                    "valid_accuracy": [],
                    "valid_f1_score": [],
                    "epoch": []
                }
    
    for metric in result_dict.keys():
        result_dict[metric].extend(results_from_experiment(event_path, metric))

    return result_dict
 ```

![Loss_plot](https://github.com/manujosephv/pytorch_tabular/assets/75081556/2e570e7f-b1ea-47a3-aa4a-9d6eb90767eb)

**Comment by manujosephv** [comment]
With default settings, train Loss is logged for each iteration, whereas val loss is logged for each epoch.

Instead of plotting yourself, can you try opening Tensorboard?
